pallets/eth-bridge/src/benchmarking.rs

```diff
@@ -164,6 +168,65 @@ fn setup_active_tx<T: Config>(
     });
 }
 
+fn setup_new_active_tx<T: Config>(
+    tx_id: EthereumId,
+    num_confirmations: u32,
+    sender: Validator<<T as pallet_avn::Config>::AuthorityId, T::AccountId>,
+) {
+    setup_active_tx::<T>(
+        tx_id,
+        num_confirmations,
+        sender,
+        BoundedVec::default(),
+        BoundedVec::default(),
+    );
+}
+
+fn setup_active_tx_with_failure_corroborations<T: Config>(
+    tx_id: EthereumId,
+    num_confirmations: u32,
+    sender: Author<T>,
+    authors: Vec<Author<T>>,
+    author: &Author<T>,
+) {
+    let mut local_authors: Vec<Author<T>> = authors.to_vec();
+    local_authors.retain(|author_from_vec| author_from_vec.account_id != sender.account_id);
+    local_authors.retain(|author_from_vec| author_from_vec.account_id != author.account_id);
+
+    let quorum = avn::Pallet::<T>::quorum() as usize;
+
+    let ( num_failure_corroborations, num_successful_corroborations ) = get_num_corroborations::<T>(quorum);
+
+    let success_authors: Vec<T::AccountId> = local_authors
+        .iter()
+        .take(num_successful_corroborations)
+        .map(|author| author.account_id.clone())
+        .collect();
+
+    let failure_authors: Vec<T::AccountId> = local_authors
+        .iter()
+        .skip(num_successful_corroborations)
+        .take(num_failure_corroborations as usize)
+        .map(|author| author.account_id.clone())
+        .collect();
+
+    setup_active_tx::<T>(
+        tx_id,
+        num_confirmations,
+        sender,
+        BoundedVec::try_from(success_authors).unwrap_or_default(),
+        BoundedVec::try_from(failure_authors).unwrap_or_default(),
+    );
+}
+
+fn get_num_corroborations<T: Config>(quorum: usize) -> (usize, usize) {
+    let num_failure_corroborations = quorum * 1/3;
+
+    let num_successful_corroborations = quorum - num_failure_corroborations - 1; // because we are adding the last
```

Comment:
```suggestion
fn get_num_corroborations<T: Config>(collators_count: usize) -> (usize, usize) {
    let quorum = avn::Pallet::<T>::quorum() as usize;

    let num_failure_corroborations = quorum - 1; // the max bad collators the system can tolerate
    let num_successful_corroborations = collators_count - num_failure_corroborations - 1; // because we are adding the last

```